### PR TITLE
NAS-116154 / 22.12 / Improve error message when quota cannot be set on user/group

### PR DESCRIPTION
**Pull request by themylogin**

### CODE REVIEW

**Comment by themylogin** [comment]
time 0:45

File: src/middlewared/middlewared/plugins/pool.py [hunk 1]
```diff
@@ -4040,14 +4040,12 @@ async def set_quota(self, ds, data):
...
                 else:
-                    id_type = ('user', 'uid') if 'user' in quota_type else ('group', 'gid')
                     xid = int(q["id"])
 
                 if xid == 0:
                     verrors.add(
                         f'quotas.{i}.id',
-                        f'{quota_type}: Setting {id_type[0]} quota on {id_type[1]} [{xid}] '
-                        'is not permitted.'
+                        f'Setting {quota_type} quota on {id_type[0]}id [{xid}] is not permitted.'
```
Comment (anodos325) [inline src/middlewared/middlewared/plugins/pool.py:4048]: id_type is now a string so we probably don't want first char here. Also space missing between `{id_type}` and "id".

Comment (themylogin) [reply src/middlewared/middlewared/plugins/pool.py:4048]: We want `uid` for `user` and `gid` for `group`, that's what `{id_type[0]}id` is

Comment (anodos325) [reply src/middlewared/middlewared/plugins/pool.py:4048]: Ah, right. got it.

**Review by anodos325 — APPROVED** [review]

**Pull request merged**